examples/text_embeddings_v2/export_v2.py

```diff
@@ -0,0 +1,187 @@
+    token, embedding = parse_line_fn(line)
+    if not embeddings_dim:
+      embeddings_dim = len(embedding)
+    elif embeddings_dim != len(embedding):
+      raise ValueError(
+          "Inconsistent embedding dimension detected, %d != %d for token %s",
+          embeddings_dim, len(embedding), token)
+
+    vocabulary.append(token)
+    embeddings.append(embedding)
+
+  return vocabulary, np.array(embeddings)
+
+
+def write_vocabulary_file(vocabulary):
+  """Write temporary vocab file for module construction."""
+  tmpdir = tempfile.mkdtemp()
+  vocabulary_file = os.path.join(tmpdir, "tokens.txt")
+  with tf.io.gfile.GFile(vocabulary_file, "w") as f:
+    for entry in vocabulary:
+      f.write(entry + "\n")
+  return vocabulary_file
+
+
+class TextEmbeddingModel(tf.train.Checkpoint):
+  """Text embedding model.
+
+  A text embeddings model that takes a sentences on input and outputs the
+  sentence embedding.
+  """
+
+  def __init__(self, vocab_file_path, oov_buckets):
+    super(TextEmbeddingModel, self).__init__()
+    self._vocabulary, self._pretrained_vectors = load(vocab_file_path,parse_line)
+    self._oov_buckets = oov_buckets
+    # Make the vocabulary file a `TrackableAsset` to ensure it is saved along with the model.
+    self._vocabulary_file = tracking.TrackableAsset(
+        write_vocabulary_file(self._vocabulary))
+    self._table = lookup_ops.index_table_from_file(
+        vocabulary_file=self._vocabulary_file,
+        num_oov_buckets=self._oov_buckets,
+        hasher_spec=lookup_ops.FastHashSpec)
+    oovs = np.zeros([oov_buckets, self._pretrained_vectors.shape[1]])
+    self._pretrained_vectors.resize(
+	[self._pretrained_vectors.shape[0] + oov_buckets, 
+	self._pretrained_vectors.shape[1]])
+    self._pretrained_vectors[self._pretrained_vectors.shape[0]:,:] = oovs
+    self.embeddings = tf.Variable(self._pretrained_vectors)
+    self.variables = [self.embeddings]
+    self.trainable_variables = self.variables
+  
+  @tf.function(input_signature=[tf.TensorSpec([None], tf.dtypes.string)])
+  def _tokenize(self, sentences):
+    # Perform a minimalistic text preprocessing by removing punctuation and
+    # splitting on spaces.
+    normalized_sentences = tf.strings.regex_replace(
+        input=sentences, pattern=r"\pP", rewrite="")
+    normalized_sentences = tf.reshape(normalized_sentences, [-1])
+    sparse_tokens = tf.strings.split(normalized_sentences, " ").to_sparse()
+
+    # Deal with a corner case: there is one empty sentence.
+    sparse_tokens, _ = tf.sparse.fill_empty_rows(sparse_tokens, tf.constant(""))
+    # Deal with a corner case: all sentences are empty.
+    sparse_tokens = tf.sparse.reset_shape(sparse_tokens)
+    sparse_token_ids = self._table.lookup(sparse_tokens.values)
+
+    return (sparse_tokens.indices, sparse_token_ids, sparse_tokens.dense_shape)
+
+  @tf.function(input_signature=[tf.TensorSpec([None], tf.dtypes.string)])
+  def __call__(self, sentences):
+    
```

Comment:
Please remove this space.

Reply:
@vbardiovskyg notes : 

1. removed that space.
2. added flags for number of lines to use and ignore. need review on those.
3. removed this line 
``    self._pretrained_vectors[self._pretrained_vectors.shape[0]:,:] = oovs``
because it creates conflict with the num_oov_buckets like this : 

![image](https://user-images.githubusercontent.com/17886829/59606169-48f63c80-9132-11e9-93a3-1dee62d4f129.png)

The line above it for resizing the array automatically adds the zeros as mentioned [here](https://docs.scipy.org/doc/numpy/reference/generated/numpy.resize.html)

`` self._pretrained_vectors.resize(
	[self._pretrained_vectors.shape[0] + oov_buckets, 
	self._pretrained_vectors.shape[1]])``

according to docs : 

``numpy.resize(a, new_shape)[source]
Return a new array with the specified shape.

If the new array is larger than the original array, then the new array is filled with repeated copies of a. Note that this behavior is different from a.resize(new_shape) which fills with zeros instead of repeated copies of a.
``

Since we are doing pretrained_vectors.resize() as many zeros as oov are being added already.